### DISCUSSION TITLE
Transform not in http context

### DISCUSSION THREAD

**Discussion by Vishwas-verma** [adonisjs/core#2662] [General]
public async login({request, auth, response, transform}: HttpContextContract) {
    try {
      const validatedData = await request.validate({
        schema: loginSchema
      })
      const user = await auth.attempt(validatedData.email, validatedData.password)
      return transform.item(user, UserTransformer)
    } catch (e) {
      response.badRequest(e.message)
    }
  }

Hye can anyone help why transform is not in HttpContextContract.
It is giving error TS2339: Property 'transform' does not exist on type 'HttpContextContract'. error

**Comment by RomainLanz** [comment]
Hey @Vishwas-verma! 👋 

Where did you saw that we provide a `transform` property inside the `HttpContext`?
📚 https://docs.adonisjs.com/guides/context#http-context-properties

**Reply by Vishwas-verma** [reply]
Hello @RomainLanz ,
It is mentioned inside this package,
https://www.npmjs.com/package/adonis-bumblebee-ts.
![B96EF072-FCC7-4CAB-A6E0-51990353AF5E](https://user-images.githubusercontent.com/41827033/119275410-8757d880-bc32-11eb-93c8-49ae3aed9df0.png)

Here is the snippet picture.

**Reply by RomainLanz** [reply]
It seems to be an issue in their library. Your error has nothing to do with AdonisJS. 😄 
https://github.com/adonisjs/docs.adonisjs.com/pull/16

**Reply by Vishwas-verma** [reply]
public async signup({request, response}: HttpContextContract) {
    try {
      // Other way using inbuilt function
      const user = await User.create(request.all())
      return response.json({
        user : await (new UserTransformer()).transform(user),
      })
    } catch (e) {
      response.badRequest(e.message)
    }
  }

I found a way to use transformer like this.
